### PR TITLE
feat(vscode): Implement parameterize existing connections command

### PR DESCRIPTION
**Pull request by ccastrotrejo**
### Main Code Changes
This pull request primarily focuses on adding functionality to parameterize connections in the VS Code extension for Azure Logic Apps. It introduces a new command to parameterize existing connections.

* A new file was added that defines the `promptParameterizeConnections` and `parameterizeConnections` functions, which handle the logic for parameterizing connections in the project.
* The `parameterizeConnections` command was registered.
* Added the `parameterizeConnections` and `parameterizeConnectionsInProjectLoadSetting` constants to manage the new functionality. 
* The `promptParameterizeConnections` function is now called during the extension activation process.
* Added the `parameterizeConnections` command to the package configuration and updated the settings schema to include the `parameterizeConnectionsInProjectLoad` setting. 
* Updated the `getParametersJson` and `saveWorkflowParameter` functions to use the updated `Parameter` type. 
* Updated the `parameterizeConnection` function to return a more specific union type instead of `any`.

---

#### Important notes
- User will be asked if they want to parameterize existing connections at project load.
- User will be asked if they want to parameterize existing connections when executing the deployment scripts command if the setting is not set to true.
- The command will only be accessible from the command palette when the `parameterizeConnectionsInProjectLoadSetting` is set to false. In case user want to execute it later.
- The functionality will parameterize all type of connections except the serviceProviderConnections

### Screenshots

https://github.com/Azure/LogicAppsUX/assets/102700317/778b27ad-dc9d-4b94-af9d-5b0d51fb7a31

Screenshot 2024-03-06 at 9 12 26 AM

Screenshot 2024-03-06 at 12 13 57 PM

### CODE REVIEW

File: apps/vs-code-designer/src/app/commands/generateDeploymentScripts/generateDeploymentScripts.ts [hunk 1]
```diff
@@ -36,8 +39,24 @@ export async function generateDeploymentScripts(context: IActionContext, project
     ext.outputChannel.appendLog(localize('initScriptGen', 'Initiating script generation...'));
 
     addLocalFuncTelemetry(context);
-    const commandIdentifier = extensionCommand.generateDeploymentScripts;
-    showPreviewWarning(commandIdentifier);
+    showPreviewWarning(extensionCommand.generateDeploymentScripts);
+    const parameterizeConnectionsSetting = getGlobalSetting(parameterizeConnectionsInProjectLoadSetting);
+
+    if (!parameterizeConnectionsSetting) {
+      const message = localize(
```
Comment (ccastrotrejo) [inline apps/vs-code-designer/src/app/commands/generateDeploymentScripts/generateDeploymentScripts.ts:46]: Hi @ecfan, could you please check the wording on this?

File: apps/vs-code-designer/src/app/commands/parameterizeConnections.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+ * Prompts the user to parameterize connections at project load.
+ * @param {IActionContext} context - The action context.
+ * @returns A promise that resolves when the operation is complete.
+ */
+export async function promptParameterizeConnections(context: IActionContext): Promise<void> {
+  if (workspace.workspaceFolders && workspace.workspaceFolders.length > 0) {
+    const workspaceFolder = await getWorkspaceFolder(context);
+    const projectPath = await tryGetLogicAppProjectRoot(context, workspaceFolder);
+    if (projectPath) {
+      const message = localize('allowParameterizeConnections', 'Allow connections to be parameterized at project load.');
```
Comment (ccastrotrejo) [inline apps/vs-code-designer/src/app/commands/parameterizeConnections.ts:30]: Hi @ecfan , could you please check the wording on this?

File: apps/vs-code-designer/src/app/commands/parameterizeConnections.ts [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+                connectionKey,
+                parametersJson,
+                localSettingsJson.Values
+              );
+            });
+          }
+        });
+        await saveWorkflowParameter(context, projectPath, parametersJson);
+        await saveConnectionReferences(context, projectPath, { connections: connectionsJson, settings: localSettingsJson.Values });
+        window.showInformationMessage(localize('finishedParameterizingConnections', 'Finished parameterizing existing connections.'));
```
Comment (ccastrotrejo) [inline apps/vs-code-designer/src/app/commands/parameterizeConnections.ts:84]: Hi @ecfan, could you please check the wording on this?

File: apps/vs-code-designer/src/package.json [hunk 4]
```diff
@@ -926,6 +936,11 @@
             "type": "boolean",
             "default": true,
             "description": "Default the data mapper to use the expanded function cards."
+          },
+          "azureLogicAppsStandard.parameterizeConnectionsInProjectLoad": {
```
Comment (ccastrotrejo) [inline apps/vs-code-designer/src/package.json:940]: Hi @ecfan, could you please check the wording in this?

File: apps/vs-code-designer/src/app/commands/generateDeploymentScripts/generateDeploymentScripts.ts (hunk 1 above)
Comment (ecfan) [reply apps/vs-code-designer/src/app/commands/generateDeploymentScripts/generateDeploymentScripts.ts:47]: I don't see any strings to review in the preceding highlighted section?

File: apps/vs-code-designer/src/app/commands/generateDeploymentScripts/generateDeploymentScripts.ts [hunk 5]
```diff
@@ -36,8 +39,24 @@ export async function generateDeploymentScripts(context: IActionContext, project
...
+        'parameterizeInDeploymentScripts',
+        'Allow connections to be parameterized? Otherwise the generation of deployment scripts will be cancelled.'
```
Comment (ecfan) [inline apps/vs-code-designer/src/app/commands/generateDeploymentScripts/generateDeploymentScripts.ts:49]: Does the following work?

```suggestion
        'Allow connection parameterization? Declining cancels deployment scripts generation.'
```

File: apps/vs-code-designer/src/app/commands/parameterizeConnections.ts (hunk 2 above)
Comment (ecfan) [inline apps/vs-code-designer/src/app/commands/parameterizeConnections.ts:30]:
```suggestion
      const message = localize('allowParameterizeConnections', 'Allow parameterization for connections at project load.');
```

File: apps/vs-code-designer/src/app/commands/parameterizeConnections.ts (hunk 3 above)
Comment (ecfan) [reply apps/vs-code-designer/src/app/commands/parameterizeConnections.ts:84]:
```suggestion
        window.showInformationMessage(localize('finishedParameterizingConnections', 'Finished parameterizing connections.'));
```

File: apps/vs-code-designer/src/package.json [hunk 6]
```diff
@@ -926,6 +936,11 @@
...
+            "type": "boolean",
+            "default": null,
+            "description": "Parameterize existing connections in project load."
```
Comment (ecfan) [inline apps/vs-code-designer/src/package.json:939]:
```suggestion
            "description": "Parameterize connections at project load."
```

**Review by ecfan — CHANGES_REQUESTED** [review]
Reviewed, thanks!

File: apps/vs-code-designer/src/app/commands/generateDeploymentScripts/generateDeploymentScripts.ts (hunk 1 above)
Comment (ccastrotrejo) [reply apps/vs-code-designer/src/app/commands/generateDeploymentScripts/generateDeploymentScripts.ts:47]: Yes, that's what I meant. Sorry about the confusion

**Review by ccastrotrejo — COMMENTED** [review]
Self-reviewed

**Pull request merged**